runtime/ast/function_declaration.go

```diff
@@ -55,7 +55,9 @@ func (d *FunctionDeclaration) Accept(visitor Visitor) Repr {
 func (d *FunctionDeclaration) Walk(walkChild func(Element)) {
 	// TODO: walk parameters
 	// TODO: walk return type
-	walkChild(d.FunctionBlock)
+	if d.FunctionBlock != nil {
```

Comment:
is it possible to move the nil checks to a common location (e.g: [ast/walk.go#L42](https://github.com/onflow/cadence/blob/master/runtime/ast/walk.go#L42)), just to be safe?

Reply:
That's a great idea and I had actually tried this, then spent a good half hour wondering why it isn't working: as the walk function parameter has type `func(Element)`, calling it with a concrete element type, e.g. `*FunctionBlock` and which is `nil`, won't be visible/detectable in the implementation of the function that is passed, in `ast.Walk`. This is because it is now a *typed nil*, and not the untyped nil used in the equality check (e.g. `== nil`). 

Here's a Go Playground that demonstrates the issue ("Go feature" ...): https://play.golang.org/p/m-dElD3ckXE

This is fairly common Go gotcha, and I wish Go didn't have this "feature"